daemon/health.go

```diff
@@ -188,12 +188,19 @@ func handleProbeResult(d *Daemon, c *container.Container, result *types.Healthch
 func monitor(d *Daemon, c *container.Container, stop chan struct{}, probe probe) {
 	probeTimeout := timeoutWithDefault(c.Config.Healthcheck.Timeout, defaultProbeTimeout)
 	probeInterval := timeoutWithDefault(c.Config.Healthcheck.Interval, defaultProbeInterval)
+loop:
 	for {
 		select {
 		case <-stop:
 			logrus.Debugf("Stop healthcheck monitoring for container %s (received while idle)", c.ID)
 			return
 		case <-time.After(probeInterval):
+			if err := d.healthcheckLimiter.Acquire(context.Background(), 1); err != nil {
```

Comment:
context has to be cancellable else the Acquire will remain stuck forever

Reply:
Which context do we use for that?

Reply:
Do we want to move the timeout up, above the acquire?

Reply:
Moved the timeout up to the top. I'm going to add metrics to track the health check runtime and waiting period.